java/test/org/openqa/selenium/chrome/ChromeOptionsTest.java

```diff
@@ -67,6 +70,53 @@ public void canBuildLogLevelFromStringRepresentation() {
     assertThat(ChromeDriverLogLevel.fromString("SEVERE")).isEqualTo(SEVERE);
   }
 
+  @Test
+  public void canAddW3CCompliantOptions() {
+    ChromeOptions chromeOptions = new ChromeOptions();
+    chromeOptions.setBrowserVersion("99")
+      .setPlatformName("9 3/4")
+      .setUnhandledPromptBehaviour(UnexpectedAlertBehaviour.IGNORE)
+      .setAcceptInsecureCerts(true)
+      .setPageLoadStrategy(PageLoadStrategy.EAGER)
+      .setStrictFileInteractability(true)
+      .setImplicitWaitTimeout(Duration.ofSeconds(1))
+      .setPageLoadTimeout(Duration.ofSeconds(2))
+      .setScriptTimeout(Duration.ofSeconds(3));
+
+    Map<String, Object> mappedOptions = chromeOptions.asMap();
+    assertThat(mappedOptions.get("browserName")).isEqualTo("chrome");
+    assertThat(mappedOptions.get("browserVersion")).isEqualTo("99");
+    assertThat(mappedOptions.get("platformName")).isEqualTo("9 3/4");
+    assertThat(mappedOptions.get("unhandledPromptBehavior").toString()).isEqualTo("ignore");
+    assertThat(mappedOptions.get("acceptInsecureCerts")).isEqualTo(true);
+    assertThat(mappedOptions.get("pageLoadStrategy").toString()).isEqualTo("eager");
+    assertThat(mappedOptions.get("strictFileInteractability")).isEqualTo(true);
+
+    Map<String, Long> expectedTimeouts = new HashMap<>();
+    expectedTimeouts.put("implicit", 1000L);
+    expectedTimeouts.put("pageLoad", 2000L);
+    expectedTimeouts.put("script", 3000L);
+
+    assertThat(expectedTimeouts).isEqualTo(mappedOptions.get("timeouts"));
+  }
+
+  @Test
+  public void canAddSequentialTimeouts() {
+    ChromeOptions chromeOptions = new ChromeOptions();
+    chromeOptions.setImplicitWaitTimeout(Duration.ofSeconds(1));
+
+    Map<String, Object> mappedOptions = chromeOptions.asMap();
+    Map<String, Long> expectedTimeouts = new HashMap<>();
+
+    expectedTimeouts.put("implicit", 1000L);
+    assertThat(expectedTimeouts).isEqualTo(mappedOptions.get("timeouts"));
+
+    chromeOptions.setPageLoadTimeout(Duration.ofSeconds(2));
+    expectedTimeouts.put("pageLoad", 2000L);
+    Map<String, Object> mappedOptions2 = chromeOptions.asMap();
+    assertThat(expectedTimeouts).isEqualTo(mappedOptions2.get("timeouts"));
```

Comment:
Perhaps:

```java
ChromeOptions chromeOptions = new ChromeOptions();
chromeOptions.setImplicitWaitTimeout(Duration.ofSeconds(1));

assertThat(chromeOptions.getCapability(TIMEOUTS)).isEqualTo(Map.of("implicit", 1000L);

chromeOptions.setPageLoadTimeout(Duration.ofSeconds(2));
assertThat(chromeOptions.getCapability(TIMEOUTS)).isEqualTo(Map.of("implicit", 1000L, "page", 2000L);
```

Reply:
There's no need to convert the options to an intermediate map.

Reply:
agreed, doesn't make sense if needing to look at existing capabilities